test/test_components.py

```diff
@@ -1046,7 +1049,7 @@ def test_component_functions(self):
         )
         dataframe_input = gr.Dataframe()
         output = dataframe_input.preprocess(x_data)
-        self.assertEqual(output[1][1], 24)
+        self.assertEqual(output.iloc[1]["Age"], 24)
```

Comment:
Left some comments explaining why on slack but `self.assertEqual(output["Age"][1], 24)`  makes the diff easier to explain and justify. Not blocking though